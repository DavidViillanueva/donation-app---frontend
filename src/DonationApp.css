.form-donation{
    width: 350px;
    height: auto;
    color: #fff;
    top: 50%;
    left: 50%;
    position: absolute;
    transform: translate(-50%,-50%);
    box-sizing: border-box;

}

.data-block{
    background: #368ae8;
    border-radius: 5px;
    padding: 10px 15px; 
    display: grid;
}

h2{
    margin-top: 10px;
    text-align: center;
}

h4{
    margin: 0px;
}

p{
    font-size: 17px;
}

small{
    cursor: pointer;
    text-align: center;
    margin-top: 5px;
    text-decoration: none;
    color: wheat;
}


.bottom{
    justify-content: center;
}

.secondary-button{
    font-size: 17px;
}

