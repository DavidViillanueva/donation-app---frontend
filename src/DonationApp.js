import React, { useState } from 'react';
import ProgressBar from './components/materialComponents/ProgressBar';
import BalloonText from './components/BalloonText';
import FormDonation from './components/FormDonation';
import DescribeCause from './components/DescribeCause';
import CopyLinkButton from './components/CopyLinkButton';

import useCounter from './hooks/useCounter';
import getDaysLeft from './helpers/getDaysLeft';

import './DonationApp.css';
import AlertNegSum from './components/AlertNegSum';

const DonationApp = ({ expireDate = new Date(), cause = '', amount = 0, doners = 0, describe='' }) => {

    //daysLeft indica la cantidad de dias faltantes para el fin de la causa
    const daysLeft = getDaysLeft( expireDate );
    //actualAmount indica la cantidad recolectada hasta ahora
    const [actualAmount, setactualAmount] = useState( 0 );
    //donationValue es la cantidad de dinero que dona el usuario
    const [donationValue, setDonationValue] = useState( 0 )
    //negativeDonation es una variable tipo bandera para mostrar o no una alerta
    const [negativeDonation, setNegativeDonation] = useState(false)
    //descriptionOpen es una variable tipo bandera para mostrar o no la descripcion de la causa
    const [descriptionOpen, setdescriptionOpen] = useState(false)
    //counterDoners mantiene la cuenta de los Donantes
    const { counter: counterDoners, increment: incrementDoners } = useCounter( doners );

    const handleSubmit = ( e ) => {
        e.preventDefault();
        if(donationValue > 0){
            //si la donacion es un valor mayor a 0
            const messageConfirm = `Are you sure you want to donate ${donationValue} to the ${cause} cause?`;
            if( window.confirm(messageConfirm) ){
                //el usuario acepta la donacion
                setactualAmount( actualAmount + parseInt (donationValue) );
                incrementDoners();
                resetInput();
            }else{
                //el usuario niega la donacion
                resetInput();
            }
        }else{
            setNegativeDonation(true);
        }
    }

    const handleChange = ({target}) => {
        setDonationValue( target.value );
    };

    const resetInput = () => {
        setDonationValue( 0 );
    }


    

    return (
        <div className="form-donation">
            
            <BalloonText 
                amount = { amount }
                actualAmount = { actualAmount }
            />

            <div className="data-block">
                
                <h2>{cause}</h2>

                <ProgressBar amount={ amount } actualAmount={ actualAmount }/>
                {negativeDonation &&
                    <AlertNegSum />
                }
                <h4>Only {daysLeft} day left to fund this project</h4>
                
                <p>
                    Join the { counterDoners } other doners who have already support this project. 
                    Every dolar helps!          
                </p>

                <FormDonation 
                    handleSubmit = { handleSubmit }
                    donationValue = { donationValue }
                    handleChange = { handleChange }
                />

                <small
                    onClick={() => setdescriptionOpen(!descriptionOpen) }
                    href='#'
                >
                    Why give ${ donationValue }?
                </small>

                <div className="row bottom">

                    <CopyLinkButton 
                        text = { () => {
                            return (window.location.href)    
                        }}
                    />
                    
                    <button className="secondary-button">
                         Go to the site!
                    </button>
                </div>
            </div>

            
            <DescribeCause 
                open = { descriptionOpen }
                describe = { describe }
            />
        </div>
    )
}

export default DonationApp;
