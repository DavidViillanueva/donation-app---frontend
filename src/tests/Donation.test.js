import React from 'react';
import { shallow } from 'enzyme';
import DonationApp from '../DonationApp'

describe('DonationApp test', () => {

    test('should render correctly', () => {
        const wrapper = shallow(<DonationApp
            expireDate = { new Date(2020,11,1)}
            cause = {'test'} 
            amount = { 10000 }
            doners = { 0 }
            describe = {'test'}
          />);
        
        expect( wrapper ).toMatchSnapshot();
    });

    
    
    
})
