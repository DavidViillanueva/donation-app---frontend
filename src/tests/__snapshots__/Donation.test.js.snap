// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonationApp test should render correctly 1`] = `
<div
  className="form-donation"
>
  <BalloonText
    actualAmount={0}
    amount={10000}
  />
  <div
    className="data-block"
  >
    <h2>
      test
    </h2>
    <ProgressBar
      actualAmount={0}
      amount={10000}
    />
    <h4>
      Only 
      6
       day left to fund this project
    </h4>
    <p>
      Join the 
      0
       other doners who have already support this project. Every dolar helps!
    </p>
    <FormDonation
      donationValue={0}
      handleChange={[Function]}
      handleSubmit={[Function]}
    />
    <small
      href="#"
      onClick={[Function]}
    >
      Why give $
      0
      ?
    </small>
    <div
      className="row bottom"
    >
      <CopyLinkButton
        text={[Function]}
      />
      <button
        className="secondary-button"
      >
        Go to the site!
      </button>
    </div>
  </div>
  <DescribeCause
    describe="test"
    open={false}
  />
</div>
`;
